pkg/crc/api/api.go

```diff
@@ -69,12 +90,22 @@ func (api CrcApiServer) handleConnection(conn net.Conn) {
 		logging.Error("Error decoding request: ", err.Error())
 		return
 	}
-
-	if handler, ok := api.handlers[req.Command]; ok {
-		result := handler(req.Args)
-		writeStringToSocket(conn, result)
+	if req.Command == "start" || req.Command == "stop" || req.Command == "delete" {
+		// queue new request to channel
+		if !addRequestToChannel(req, api.clusterOpsRequestsChan) {
+			defer conn.Close()
```

Comment:
better to use a new struct holding both the request and the connection, and to add it to a single channel. Having 2 channels which must have exactly the same number of items could break easily if we don't pay attention.

Reply:
Yes, i was also considering this.. thanks!